src/main/java/school/lemon/changerequest/java/matrix/Matrix.java

```diff
@@ -0,0 +1,104 @@
+package school.lemon.changerequest.java.matrix;
+
+/**
+ * Created by Yaroslav Pavlinskiy on 12.12.2016.
+ */
+public class Matrix {
+
+    private final int Rows; // count Rows
+    private final int Columns; // count Columns
+    private final int[][] data; // 2d array for matrix
+
+    public Matrix(int R,int C)// Constructor with Rows and Columns
+    {
+        this.Rows = R;
+        this.Columns = C;
+        this.data = new int[R][C];
+    }
+
+    public Matrix(int[][]data) //Constructor 2d array
+    {
+        this.Rows = data.length;
+        this.Columns = data[0].length;
+        this.data = data;
+    }
+    private Matrix(Matrix A) { this(A.data); }
+
+    public Matrix AddMatrix(Matrix B) // Method that adding matrix A and B
+    {
+        return Calculate(B,'+');
+    }
+
+    public Matrix SubtractMatrix(Matrix B)
+    {
+        return Calculate(B,'-');
+    }
+
+    public Matrix MultiplyNumber(int n)
+    {
+        Matrix A = this;
+        Matrix C = new Matrix(Rows, Columns);
+        for (int i = 0; i < Rows; i++)
+            for (int j = 0; j < Columns; j++)
+                C.data[i][j] = A.data[i][j] * n;
+        return C;
+    }
+
+    public  Matrix MultiplyMatrix(Matrix B)
+    {
+        return Calculate(B,'*');
+    }
+
+    public Matrix Transpose()
+    {
+        Matrix A = this;
+        Matrix C = new Matrix(Rows,Columns);
+        for (int i = 0; i < Rows; i++)
+            for (int j = 0; j < Columns; j++)
+                C.data[i][j] = A.data[j][i];
+        return C;
+    }
+
+    public Matrix Calculate(Matrix B,char d)
```

Comment:
1. There is no need in this method at all. Determining operation by `char` is not the good approach, because you now which method was called and which operation to execute. So please split your logic to specific methods.
2. Not sure multiplication is done correctly. Please read about it https://en.wikipedia.org/wiki/Matrix_multiplication
3. As you may remember the point was that current matrix should be changed. So no need to create third matrix. You should modify existing one.